modules/puppetca_token_whitelisting/puppetca_token_whitelisting_autosigner.rb

```diff
@@ -0,0 +1,123 @@
+require 'jwt'
+require 'openssl'
+
+module ::Proxy::PuppetCa::TokenWhitelisting
+  class Autosigner
+    include ::Proxy::Log
+    include ::Proxy::Util
+
+    JWT_ALGORITHM = 'RS512'
+
+    def tokens_file
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.tokens_file
+    end
+
+    def sign_all
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.sign_all
+    end
+
+    def smartproxy_cert
+      @certificate ||= OpenSSL::PKey::RSA.new File.read cert_file
+    end
+
+    def storage
+      Proxy::PuppetCa::TokenWhitelisting::TokenStorage.new tokens_file
+    end
+
+    def token_ttl
+      Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.token_ttl
+    end
+
+    def cert_file
+      return Proxy::SETTINGS.ssl_private_key.to_s if Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.certificate.nil?
+
+      file = Proxy::PuppetCa::TokenWhitelisting::Plugin.settings.certificate
+      unless File.exist?(file)
+        File.write file, OpenSSL::PKey::RSA.generate(2048)
```

Comment:
It's probably good to do this with a umask so it's not world readable (or some other way). The number of bits should also be at least a constant IMHO.

Reply:
Added.